.github/contributing.md

```diff
@@ -0,0 +1,245 @@
+
+> NOTE: Manual option is only available after PR creation from `Development` sidebar option.
+
+If unsure about whether PR is ready for review, create a draft PR.<br>
+Update PR to be full PR and ask for reviewers when ready.
+
+If changes are requested,
+1. Ask for clarification on changes necessary, if needed.
+2. Don't be afraid to state your point of view on proposed changes.
+3. Debate respectfully, acknowledge the other contributors point of view
+4. Finalize decision and solution to identify whether changes are actually necessary
+- If yes, make changes locally. <br>Follow <span style="color: #5C9340;">## Pre-committing to Github</span> guidelines defined above and push changes to Github
+5. Request re-review from contributors once ready again.
+
+> NOTE: Every PR __must__ require at least 2 approvals before merge-ready.
+
+
+<br>
+
+If changes elsewhere are pushed to the `dev` branch, your branch will need to pick up these new changes before adding your changes on top.
+
+This inevitably means any branches that are at PR or review stage will need to be rebased with `dev` before merging
+
+1. Navigate to the local branch you are completing the changes upon.
+2. Execute `git rebase dev`.
+3. Observe potential merge conflicts
+- If yes, resolve merge conflicts and execute `git rebase --continue` to resume the merge of `dev` new code into your branch
+- If no, message indicates `Successfully rebased with dev` in CLI
+
+<br>
+
+
+## PR Review process
+
+Upon request, a contributor is obligate to review another contributor's PR.<br>
+The following guidelines need to be followed upon undertaking a review.
+
+ Review should be:
+ - Thorough
+ - Articulate
+ - Respectful
+ - Split in two parts:
+
+    Single Line Review Comments should consider:
+    1. Syntax changes needed
+    2. Misunderstanding of code provided
+    3. Formatting adjustments required
+    4. Additions/suggestions for improvements in code
+
+    Overall Review Comment should consider:
+    1. Switch to branch and execute flow targeted by PR
+    2. Observe physical breakages/issues - Report as comment
+    3. Execute all tests surrounding the PR
+    4. Observe failures - Report as comment
+    5. Any edge cases contributor didn't consider - Report in comment
+    6. Overall review sentiment for the contributors
+    7. Next steps
+
+
+ - a structured format for a potential debate
+
+ {Marco and Dario input here}
+
+<br>
+
+---
+
+
+Do not take abrupt change requests to heart, we are all striving to create the most elite, elegant, performant, secure solution possible for our codebase.<br>
+
+You must understand that reviewing PR's is a time-consuming process and contributors might be sharp and to the point in an effort to save time.
+
+However, just know every contribution and opinion is valuable and appreciated.
+
+🥰⏳
+
+---
+
+<br>
+
+
+
+## Merging and deploying
+
+{Dario input here}
```

Comment:
Once a PR has been approved by the reviewers, it will be merged into the target and the remote branch will be deleted.